### ISSUE TITLE
nan spring constant

### ISSUE THREAD

**Issue by pmrv** [ICAMS/calphy#92]
I've just run into an issue where the this line creates a nan, which then borks the whole calculation.

https://github.com/ICAMS/calphy/blob/6a85e668a728d8dd9840a7dd0863713b73b60f42/calphy/solid.py#L114C54-L114C54

I've attached the tarball of the pyiron job that ran into this, but I don't have much time right now to go in depth (I imagine you as well), so I just want to have brought up the issue.

Calphy version was 1.2.12.

[test0.tar.gz](https://github.com/ICAMS/calphy/files/13165375/test0.tar.gz)

(This is actually a tar.bz2 file, but github sucks.)

**Comment by pmrv** [comment]
Relevant calphy log file.

```
2023-10-25 12:59:34,666 calphy.helpers INFO     ---------------input file----------------
2023-10-25 12:59:34,667 calphy.helpers INFO     commented out as causes crash when we're expanding the T range after a fail run
2023-10-25 12:59:34,667 calphy.helpers INFO     ------------end of input file------------
2023-10-25 12:59:34,667 calphy.helpers INFO     Temperature start: 50.000000 K, temperature stop: 1000.000000 K, pressure: 0.000000 bar
2023-10-25 12:59:34,667 calphy.helpers INFO     Pressure adjusted in iso
2023-10-25 12:59:34,667 calphy.helpers INFO     Reference phase is solid
2023-10-25 12:59:34,667 calphy.helpers INFO     Thermostat/Barostat combo for equilibration cycle is not explicitely specified
2023-10-25 12:59:34,667 calphy.helpers INFO     Thermostat/Barostat combo for equilibration cycle can be specified using keyword equilibration_control
2023-10-25 12:59:34,667 calphy.helpers INFO     Reference phase is solid, setting Nose-Hoover thermostat for equilibration
2023-10-25 12:59:34,667 calphy.helpers INFO     Nose-Hoover thermostat damping is 0.100000
2023-10-25 12:59:34,667 calphy.helpers INFO     Nose-Hoover barostat damping is 0.100000
2023-10-25 12:59:34,667 calphy.helpers INFO     These values can be tuned by adding in the input file:
2023-10-25 12:59:34,667 calphy.helpers INFO     nose_hoover:
2023-10-25 12:59:34,667 calphy.helpers INFO        thermostat_damping: <float>
2023-10-25 12:59:34,667 calphy.helpers INFO        barostat_damping: <float>
2023-10-25 12:59:34,667 calphy.helpers INFO     Integration stage is done using Nose-Hoover thermostat and barostat when needed
2023-10-25 12:59:34,667 calphy.helpers INFO     Thermostat damping is 0.500000
2023-10-25 12:59:34,667 calphy.helpers INFO     Barostat damping is 0.100000
2023-10-25 12:59:37,602 calphy.helpers INFO     Lattice: file with a=1.000000
2023-10-25 12:59:37,602 calphy.helpers INFO     336 atoms in the unit cell
2023-10-25 12:59:37,602 calphy.helpers INFO     concentration:
2023-10-25 12:59:37,602 calphy.helpers INFO     [1.]
2023-10-25 12:59:37,602 calphy.helpers INFO     Input structure is read in from a LAMMPS data file
2023-10-25 12:59:37,603 calphy.helpers INFO     336 atoms in 1 cells on 1 cores
2023-10-25 12:59:37,603 calphy.helpers INFO     pair_style: meam 
2023-10-25 12:59:37,603 calphy.helpers INFO     pair_coeff: * * MgAlZn.library.meam Mg Al Zn MgAlZn.parameter.meam Mg Al Zn
2023-10-25 13:03:22,488 calphy.helpers INFO     At count 1 mean pressure is -1.221710 with 16.713732 vol/atom
2023-10-25 13:05:14,940 calphy.helpers INFO     At count 2 mean pressure is -0.772107 with 16.713594 vol/atom
2023-10-25 13:07:07,465 calphy.helpers INFO     At count 3 mean pressure is 0.469142 with 16.713719 vol/atom
2023-10-25 13:07:07,465 calphy.helpers INFO     finalized vol/atom 16.713719 at pressure 0.469142
2023-10-25 13:07:07,465 calphy.helpers INFO     Avg box dimensions x: 17.107000, y: 17.284000, z:18.994000
2023-10-25 13:08:59,972 calphy.helpers INFO     At count 1 mean k is inf std is nan
2023-10-25 13:10:52,291 calphy.helpers INFO     At count 2 mean k is inf std is nan
2023-10-25 13:12:44,792 calphy.helpers INFO     At count 3 mean k is inf std is nan
2023-10-25 13:14:37,556 calphy.helpers INFO     At count 4 mean k is inf std is nan
2023-10-25 13:16:30,319 calphy.helpers INFO     At count 5 mean k is inf std is nan
2023-10-25 13:18:22,749 calphy.helpers INFO     At count 6 mean k is inf std is nan
2023-10-25 13:20:14,855 calphy.helpers INFO     At count 7 mean k is inf std is nan
```

**Comment by srmnitc** [comment]
This is the same issue as #60

**Comment by srmnitc** [comment]
fixed in #108
